Use kube-system namespace

huggsboson opened the pull request:
Fixes #23153.

Sadly, kube-system isn't automatically created, so people need to make
sure to create it in their turnup scripts.  Also after creating
kube-system it can take 10+ seconds for master and proxy to show up.

I tested the equivalent of these changes locally, but not these changes
themselves as I don't have a dev/build env up, so please read carefully
and maybe try them out!

Commit:
Fixes #23153.

Sadly, kube-system isn't automatically created, so people need to make
sure to create it in their turnup scripts.  Also after creating
kube-system it can take 10+ seconds for master and proxy to show up.

I tested the equivalent of these changes locally, but not these changes
themselves as I don't have a dev/build env up, so please read carefully
and maybe try them out!

Use kubectl create ns

huggsboson commented:
I signed it!

luxas commented on cluster/images/hyperkube/turnup.sh:
```diff
@@ -41,3 +41,17 @@ docker run \
+    sleep 1
+done
+
+kubectl create -f - << EOF
```
`kubectl create ns kube-system`

huggsboson commented:
@luxas done.

colemickens commented:
@huggsboson The `kubernetes` service remains in the default namespace. (per @thockin in Slack). So your last commit is not correct.

huggsboson commented:
Fixed.  I'm obviously not really testing. 

Here was the chat log:
**colemickens** @huggsboson: so, weird issue here. The DNS pod fails when I changed the nslookup to use kube-system instead of default.  And, the kubernetes service shows up with kubectl get services, as in, it's in the default namespace still..
**thockin**: colemickens: the `kubernetes` service is supposed to be in `default`

mikedanese commented:
Please squash.

huggsboson commented:
squashed.

luxas commented:
ping @mikedanese

mikedanese commented on cluster/images/hyperkube/turnup.sh:
```diff
@@ -20,7 +20,7 @@ set -o errexit
 set -o pipefail
 
-K8S_VERSION=${K8S_VERSION:-"1.2.0-alpha.7"}
+K8S_VERSION=${K8S_VERSION:-"1.2.0"}
```
This is doomed to be out of dat forever. We should avoid embedding k8s_version into scripts and instead discover the most recent
